tracking/src/main/java/org/motechproject/nms/tracking/service/CacheService.java

```diff
@@ -0,0 +1,5 @@
+package org.motechproject.nms.tracking.service;
+
+public interface CacheService {
```

Comment:
@Cacheable won't work on internal methods.

http://stackoverflow.com/questions/12115996/spring-cache-cacheable-method-ignored-when-called-from-within-the-same-class

On Wed, Sep 16, 2015 at 11:07 AM, Rob LaRubbio <notifications@github.com>
wrote:

> In
> tracking/src/main/java/org/motechproject/nms/tracking/service/CacheService.java
> <https://github.com/motech-implementations/mim/pull/590#discussion_r39664464>
> :
>
> > @@ -0,0 +1,5 @@
> > +package org.motechproject.nms.tracking.service;
> > +
> > +public interface CacheService {
>
> Is there a reason you created the cache service bean? My concern is if we
> want to cache the results of another method we should just be able to add
> the @Cacheable annotation and not need to add a new method to this bean
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/pull/590/files#r39664464>.
>



-- 
Frank Huster
Mobile Health Innovations
Grameen Foundation
+1.206.325.6690 x201